### PR TITLE
Refactor failWorkflowTask method

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Rename `failCommand` to `failWorkflowTask` because it fails workflow task not command and extract other methods to better code reuse.

**Why?**
To reuse code to fail workflow task in different places.

**How did you test it?**
Existing tests.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: service/history/workflowTaskHandler.go [hunk 1]
```diff
@@ -1423,49 +1411,68 @@ func (handler *workflowTaskHandlerImpl) validateCommandAttr(
...
 ) error {
 
-	if failedCause, err := validationFn(); err != nil {
-		if _, ok := err.(*serviceerror.InvalidArgument); ok {
-			return handler.failCommand(failedCause, err)
-		}
+	return handler.failWorkflowTaskOnInvalidArgument(validationFn())
+}
+
+func (handler *workflowTaskHandlerImpl) failWorkflowTaskOnInvalidArgument(
```
Comment (yiminc) [inline service/history/workflowTaskHandler.go:1417]: why we need this method? can we just call failWorkflowTask(wtFailedCause, err) in places where you need this method?

**Review by rodrigozhou — COMMENTED** [review]
Are you not changing the error handling for `UnaliasFields` call?

File: service/history/workflowTaskHandler.go (hunk 1 above)
Comment (alexshtin) [reply service/history/workflowTaskHandler.go:1417]: I don't want to change any logic in "refactor" style PR. But yes, having both of them is not right. I was thinking about reverse approach: calling `failWorkflowTaskOnInvalidArgument` everywhere where `failWorkflowTask` is called now. Wrong "user" input should fail WT but any server errors should not. I need to track all possible errors here to make a call and it is out of scope for this small refactoring.

File: service/history/workflowTaskHandlerCallbacks.go [hunk 2]
```diff
@@ -453,7 +453,10 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 	hasPendingUpdates := ms.UpdateRegistry().HasPending(request.GetMessages())
 	hasBufferedEvents := ms.HasBufferedEvents()
 	if err := namespaceEntry.VerifyBinaryChecksum(request.GetBinaryChecksum()); err != nil {
-		wtFailedCause = NewWorkflowTaskFailedCause(enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_BINARY, serviceerror.NewInvalidArgument(fmt.Sprintf("binary %v is already marked as bad deployment", request.GetBinaryChecksum())))
+		wtFailedCause = newWorkflowTaskFailedCause(
+			enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_BINARY,
+			serviceerror.NewInvalidArgument(fmt.Sprintf("binary %v is already marked as bad deployment", request.GetBinaryChecksum())),
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply service/history/workflowTaskHandlerCallbacks.go]: and I believe word "already" doesn't belong here.

**Comment by alexshtin** [comment]
> You're not changing the error handling for `UnaliasFields` call?

Not in this one. Stay tuned :-)

**Pull request merged**